reflex-dom-plaid/Reflex/Dom/Plaid/Link.hs

```diff
@@ -131,21 +131,31 @@ activatePlaidLinkDialog cfg onResult = do
     pure ()
     )
 
+  o ^. jss (t_ "onEvent") (fun $ \_ _ args -> case args of
+      (eventNameJSVal : metaJs : _) -> do
+        eventName <- filter (/= '"') . show <$> fromJSValUnchecked @JSString eventNameJSVal
+        case eventName of
+          "EXIT" -> handleError o metaJs
+          "ERROR" -> handleError o metaJs
+          _ -> pure ()
```

Comment:
I initially didn't understand why we case, do the same thing for two cases, and do nothing for the rest.  Maybe a helper like with some haddocks explaining the two cases will help anyone in the future:
```haskell
-- | onEvent can return "Foo", "Bar", "EXIT", and "ERROR".  Exit and Error strings are errors stemming from someFoo and should be returned to the caller.
whenErrorM o m err 
  | err == "EXIT || err == "ERROR" = handleError o m 
  | otherewise = pure ()
```

Reply:
Since I don't know if we need to process other events than 'EXIT' and 'ERROR' I've picked  `case eventName of` because it's just 1 level of nesting and it's simple to extend. I've tried to use `ViewPatterns` like this:

```Haskell
{-# LANGUAGE ViewPatterns #-}
{-# OPTIONS_GHC -Wno-overlapping-patterns #-}

let ifError = flip elem ["EXIT", "ERROR"]
case eventName of
  (ifError -> _) -> handleError o metaJs
  _ -> pure ()
 ```
 
But compiler started to complain about redundant pattern and I didn't understand why. Also I've found such equations [Here](https://ghc.gitlab.haskell.org/ghc/doc/users_guide/exts/view_patterns.html):

> Matching: To the equations in Section 3.17.3 of the [Haskell 98 Report](https://www.haskell.org/onlinereport/), add the following:
> 
> case v of { (e -> p) -> e1 ; _ -> e2 }
>  =
> case (e v) of { p -> e1 ; _ -> e2 }

It started to take more time than it's worth in my opinion and I've picked the most dumb approach.

Reply:
Ok, my suggested helper is not that much of an improvment but some context in comments would be helpful explaining why we only look at "EXIT" and "ERROR";  Are these part of a specification somewhere?  Just what you've observed during debugging?  I just want some guide posts for when something upstream changes and we need to inspect the other cases to fix the breakage.